// ionic-angular/components/nav/nav-push.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
import * as import0 from 'ionic-angular/components/nav/nav-push';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
export var Wrapper_NavPush = (function () {
    function Wrapper_NavPush(p0) {
        this.changed = false;
        this._navPush = import1.UNINITIALIZED;
        this._navParams = import1.UNINITIALIZED;
        this.context = new import0.NavPush(p0);
    }
    Wrapper_NavPush.prototype.check_navPush = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._navPush, currValue))) {
            this.changed = true;
            this.context.navPush = currValue;
            this._navPush = currValue;
        }
    };
    Wrapper_NavPush.prototype.check_navParams = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._navParams, currValue))) {
            this.changed = true;
            this.context.navParams = currValue;
            this._navParams = currValue;
        }
    };
    Wrapper_NavPush.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_NavPush;
}());
export var Wrapper_NavPushAnchor = (function () {
    function Wrapper_NavPushAnchor(p0, p1) {
        this.changed = false;
        this.context = new import0.NavPushAnchor(p0, p1);
    }
    Wrapper_NavPushAnchor.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_NavPushAnchor;
}());
