// ionic-angular/components/toggle/toggle.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ionic-angular/components/toggle/toggle';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import5 from '@angular/core/src/linker/element';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from 'ionic-angular/util/form';
import * as import9 from 'ionic-angular/config/config';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from 'ionic-angular/util/haptic';
import * as import12 from 'ionic-angular/components/item/item';
import * as import13 from '@angular/forms/src/directives/control_value_accessor';
import * as import14 from '@angular/core/src/metadata/view';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from '../button/button.ngfactory';
import * as import17 from 'ionic-angular/components/button/button';
export var Wrapper_Toggle = (function () {
    function Wrapper_Toggle(p0, p1, p2, p3, p4, p5) {
        this.changed = false;
        this._color = import1.UNINITIALIZED;
        this._mode = import1.UNINITIALIZED;
        this._checked = import1.UNINITIALIZED;
        this._disabled = import1.UNINITIALIZED;
        this.context = new import0.Toggle(p0, p1, p2, p3, p4, p5);
    }
    Wrapper_Toggle.prototype.check_color = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._color, currValue))) {
            this.changed = true;
            this.context.color = currValue;
            this._color = currValue;
        }
    };
    Wrapper_Toggle.prototype.check_mode = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._mode, currValue))) {
            this.changed = true;
            this.context.mode = currValue;
            this._mode = currValue;
        }
    };
    Wrapper_Toggle.prototype.check_checked = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._checked, currValue))) {
            this.changed = true;
            this.context.checked = currValue;
            this._checked = currValue;
        }
    };
    Wrapper_Toggle.prototype.check_disabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._disabled, currValue))) {
            this.changed = true;
            this.context.disabled = currValue;
            this._disabled = currValue;
        }
    };
    Wrapper_Toggle.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_Toggle;
}());
var renderType_Toggle_Host = null;
var _View_Toggle_Host0 = (function (_super) {
    __extends(_View_Toggle_Host0, _super);
    function _View_Toggle_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Toggle_Host0, renderType_Toggle_Host, import7.ViewType.HOST, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    Object.defineProperty(_View_Toggle_Host0.prototype, "_NG_VALUE_ACCESSOR_0_5", {
        get: function () {
            if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) {
                (this.__NG_VALUE_ACCESSOR_0_5 = [this._Toggle_0_4.context]);
            }
            return this.__NG_VALUE_ACCESSOR_0_5;
        },
        enumerable: true,
        configurable: true
    });
    _View_Toggle_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-toggle', rootSelector, null);
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Toggle0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Toggle_0_4 = new Wrapper_Toggle(this.parentInjector.get(import8.Form), this.parentInjector.get(import9.Config), new import10.ElementRef(this._el_0), this.renderer, this.parentInjector.get(import11.Haptic), this.parentInjector.get(import12.Item, null));
        this._appEl_0.initComponent(this._Toggle_0_4.context, [], compView_0);
        compView_0.create(this._Toggle_0_4.context, this.projectableNodes, null);
        this._expr_0 = import1.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Toggle_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.Toggle) && (0 === requestNodeIndex))) {
            return this._Toggle_0_4.context;
        }
        if (((token === import13.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) {
            return this._NG_VALUE_ACCESSOR_0_5;
        }
        return notFoundResult;
    };
    _View_Toggle_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._Toggle_0_4.detectChangesInternal(this, this._el_0, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Toggle_0_4.context.ngAfterContentInit();
            }
        }
        var currVal_0 = this._Toggle_0_4.context._disabled;
        if (import2.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_0, 'toggle-disabled', currVal_0);
            this._expr_0 = currVal_0;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Toggle_Host0.prototype.destroyInternal = function () {
        this._Toggle_0_4.context.ngOnDestroy();
    };
    return _View_Toggle_Host0;
}(import3.AppView));
function viewFactory_Toggle_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Toggle_Host === null)) {
        (renderType_Toggle_Host = viewUtils.createRenderComponentType('', 0, import14.ViewEncapsulation.None, [], {}));
    }
    return new _View_Toggle_Host0(viewUtils, parentInjector, declarationEl);
}
export var ToggleNgFactory = new import15.ComponentFactory('ion-toggle', viewFactory_Toggle_Host0, import0.Toggle);
var styles_Toggle = [];
var renderType_Toggle = null;
var _View_Toggle0 = (function (_super) {
    __extends(_View_Toggle0, _super);
    function _View_Toggle0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Toggle0, renderType_Toggle, import7.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_Toggle0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'toggle-icon');
        this._el_1 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'toggle-inner');
        this._el_2 = this.renderer.createElement(parentRenderNode, 'button', null);
        this.renderer.setElementAttribute(this._el_2, 'class', 'item-cover');
        this.renderer.setElementAttribute(this._el_2, 'ion-button', 'item-cover');
        this.renderer.setElementAttribute(this._el_2, 'role', 'checkbox');
        this.renderer.setElementAttribute(this._el_2, 'type', 'button');
        this._appEl_2 = new import5.AppElement(2, null, this, this._el_2);
        var compView_2 = import16.viewFactory_Button0(this.viewUtils, this.injector(2), this._appEl_2);
        this._Button_2_4 = new import16.Wrapper_Button(null, 'item-cover', this.parentInjector.get(import9.Config), new import10.ElementRef(this._el_2), this.renderer);
        this._appEl_2.initComponent(this._Button_2_4.context, [], compView_2);
        compView_2.create(this._Button_2_4.context, [[]], null);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this.init([], [
            this._el_0,
            this._el_1,
            this._el_2
        ], [], []);
        return null;
    };
    _View_Toggle0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import17.Button) && (2 === requestNodeIndex))) {
            return this._Button_2_4.context;
        }
        return notFoundResult;
    };
    _View_Toggle0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._Button_2_4.detectChangesInternal(this, this._el_2, throwOnChange)) {
            this._appEl_2.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Button_2_4.context.ngAfterContentInit();
            }
        }
        var currVal_0 = this.context._checked;
        if (import2.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_0, 'toggle-checked', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.context._activated;
        if (import2.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'toggle-activated', currVal_1);
            this._expr_1 = currVal_1;
        }
        var currVal_2 = this.context.id;
        if (import2.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_2, 'id', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = this.context._checked;
        if (import2.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementAttribute(this._el_2, 'aria-checked', ((currVal_3 == null) ? null : currVal_3.toString()));
            this._expr_3 = currVal_3;
        }
        var currVal_4 = this.context._labelId;
        if (import2.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementAttribute(this._el_2, 'aria-labelledby', ((currVal_4 == null) ? null : currVal_4.toString()));
            this._expr_4 = currVal_4;
        }
        var currVal_5 = this.context._disabled;
        if (import2.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementAttribute(this._el_2, 'aria-disabled', ((currVal_5 == null) ? null : currVal_5.toString()));
            this._expr_5 = currVal_5;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Toggle0;
}(import3.AppView));
export function viewFactory_Toggle0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Toggle === null)) {
        (renderType_Toggle = viewUtils.createRenderComponentType('', 0, import14.ViewEncapsulation.None, styles_Toggle, {}));
    }
    return new _View_Toggle0(viewUtils, parentInjector, declarationEl);
}
